listenbrainz/webserver/decorators.py

```diff
@@ -12,6 +12,8 @@ def crossdomain(origin='*', methods=None, headers=None,
     # Based on snippet by Armin Ronacher located at http://flask.pocoo.org/snippets/56/.
     if methods is not None:
         methods = ', '.join(sorted(x.upper() for x in methods))
+    default_headers = {"AUTHORIZATION", "CONTENT-TYPE"}
+    headers.update(default_headers)
     if headers is not None and not isinstance(headers, string_types):
         headers = ', '.join(x.upper() for x in headers)
```

Comment:
This is a bit incorrect because headers can be None so that'll cause an error. The other way around i.e. `default_headers.update(headers)` can work because default_headers is never None. But then passing None to update can cause error.

I am almost sure this change here
```suggestion
    all_headers = {"AUTHORIZATION", "CONTENT-TYPE"}
    if headers is not None:
        all_headers.update(headers)
    all_headers = ', '.join(x.upper() for x in headers)
```

, removing the redundant check at L43 and changing L44 in the file to use `all_headers` variable should work.